syft/execution/protocol.py

```diff
@@ -33,22 +33,28 @@ class func2protocol(object):
     This class should be used only as a decorator.
     """
 
-    def __init__(self, args_shape=None, state=None):
+    def __init__(self, args_shape=None):
         self.args_shape = args_shape
 
     def __call__(self, protocol_function):
+        # create the roles present in decorator
+        roles = {role_id: Role() for role_id in self.args_shape.keys()}
+
         protocol = Protocol(
             name=protocol_function.__name__,
             forward_func=protocol_function,
+            roles=roles,
             id=sy.ID_PROVIDER.pop(),
             owner=sy.local_worker,
         )
 
         # Build the protocol automatically
+        # TODO We can always build automatically, can't we? Except if workers doesn't have
+        # tensors yet in store. Do we handle that?
```

Comment:
Yeah, I think we can always build automatically, because we create tensors to be used as inputs, which could be loaded into the (temporary) `VirtualWorkers'` stores somewhere around here.